tests/backup/backup_helper_test.go

```diff
@@ -0,0 +1,1205 @@
+			log.Infof("Restore status - %s", restoreResponseStatus)
+			log.InfoD("Status of %s - [%s]",
+				restoreName, restoreResponseStatus.GetStatus())
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("expected status of %s - [%s] or [%s], but got [%s]",
+			restoreName, api.RestoreInfo_StatusInfo_PartialSuccess.String(), api.RestoreInfo_StatusInfo_Success, restoreResponseStatus.GetStatus())
+	}
+	task.DoRetryWithTimeout(restoreSuccessCheck, 10*time.Minute, 30*time.Second)
+	return nil
+}
+
+// CreateRestore creates restore
+func CreateRestoreWithUID(restoreName string, backupName string, namespaceMapping map[string]string, clusterName string,
+	orgID string, ctx context.Context, storageClassMapping map[string]string, backupUID string) error {
+
+	backupDriver := Inst().Backup
+	log.Infof("Getting the UID of the backup needed to be restored")
+
+	createRestoreReq := &api.RestoreCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  restoreName,
+			OrgId: orgID,
+		},
+		Backup:              backupName,
+		Cluster:             clusterName,
+		NamespaceMapping:    namespaceMapping,
+		StorageClassMapping: storageClassMapping,
+		BackupRef: &api.ObjectRef{
+			Name: backupName,
+			Uid:  backupUID,
+		},
+	}
+	_, err := backupDriver.CreateRestore(ctx, createRestoreReq)
+	if err != nil {
+		return err
+	}
+	restoreInspectRequest := &api.RestoreInspectRequest{
+		Name:  restoreName,
+		OrgId: orgID,
+	}
+	restoreSuccessCheck := func() (interface{}, bool, error) {
+		resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+		if err != nil {
+			return "", false, err
+		}
+		restoreResponseStatus := resp.GetRestore().GetStatus()
+		if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess || restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Success {
+			log.Infof("Restore status - %s", restoreResponseStatus)
+			log.InfoD("Status of %s - [%s]",
+				restoreName, restoreResponseStatus.GetStatus())
+			return "", false, nil
+		}
+		return "", true, fmt.Errorf("expected status of %s - [%s] or [%s], but got [%s]",
+			restoreName, api.RestoreInfo_StatusInfo_PartialSuccess.String(), api.RestoreInfo_StatusInfo_Success, restoreResponseStatus.GetStatus())
+	}
+	task.DoRetryWithTimeout(restoreSuccessCheck, 10*time.Minute, 30*time.Second)
+	return nil
+}
+
+// TearDownBackupRestoreSpecific deletes backups and restores specified by name as well as backup location
+func TearDownBackupRestoreSpecific(backups []string, restores []string) error {
+	for _, backupName := range backups {
+		backupUID, err := getBackupUID(backupName, OrgID)
+		if err != nil {
+			return err
+		}
+		ctx, err := backup.GetAdminCtxFromSecret()
+		if err != nil {
+			return err
+		}
+		_, err = DeleteBackup(backupName, backupUID, OrgID, ctx)
+		if err != nil {
+			return err
```

Comment:
do not return err if you want to handle the failures using dash else return err at all the places, just to be consistent

Reply:
done